students/shibin_mathew/lesson3/a_new_file.py

```diff
@@ -0,0 +1,84 @@
+def exchange_first_last(seq):
+	#Exchange first and last item
+	first = ''
+	middle = ''
+	last = ''
+	tup_a = ()
+	if isinstance(seq, str):
+		return seq[-1]+seq[1:-1]+seq[0]
+	else: 
+		first = seq[-1]
+		middle = seq[1:-1]
+		last = seq[0]
+		tup_a =  tup_a +(first,)
+		tup_a = tup_a + (middle)
+		tup_a = tup_a + (last,)
+		return tup_a
+
+
+def every_other_removed(seq):
+	#Every other item removed
+	tup_b=()
+	str_length = len(seq)
+	new_str = ''
+	if isinstance(seq, str): #if sequence is a string
+		for i, item in enumerate(seq): #loop over the sequence and extract every other value
+			if i % 2 == 0: #even value
+				new_str+=item #concatenate every other values
+		return new_str #return new string
+	else:
+		for i, item in enumerate(seq):
+			if i%2 ==0:
+				tup_b = tup_b+(item,)
+		return tup_b
+
+
+def first_last_four_removed(seq):
+	#First 4 and last 4 items removed, and then every other item in between
+	str_ = ''
+	str_ = seq[4:-4]
+	str_ = str_[0:-1:2]
+	return str_
+
+
+
+def elements_reversed(seq):
+	#elements reversed
+	str_ = ""
+	tup = ()
+	
+	if isinstance(seq, str):
+		for i in range(len(seq)-1,-1,-1):
+			str_+=seq[i]
+		return str_
+	else:
+		for i in range(len(seq)-1,-1,-1):
+			tup = tup + (seq[i],)
+		return tup
```

Comment:
all of the above functions should be rewritten to use _slicing_ only (`[start:end:step]`)

the solutions for all of them are actually very simple thanks to python slicing functionality; they are simple one liners that just work for string, tuple and list without any special logic.